framework/views/pullHook.js

```diff
@@ -0,0 +1,333 @@
+        }
+
+        // Hack to make it work on Android 4.4 WebView. Scrolls manually near the top of the page so
+        // there will be no inertial scroll when scrolling down. Allowing default scrolling will
+        // kill all 'touchmove' events.
+        var el = this._pageElement[0];
+        el.scrollTop = this._startScroll - event.gesture.deltaY;
+        if (el.scrollTop < window.innerHeight && event.gesture.direction !== 'up') {
+          event.gesture.preventDefault();
+        }
+
+        if (this._currentTranslation === 0 && this._getCurrentScroll() === 0) {
+          this._transitionDragLength = event.gesture.deltaY;
+
+          var direction = event.gesture.interimDirection;
+          if (direction === 'down') {
+            this._transitionDragLength -= 1;
+          }
+          else {
+            this._transitionDragLength += 1;
+          }
+        }
+
+        var scroll = event.gesture.deltaY - this._startScroll;
+
+        scroll = Math.max(scroll, 0);
+
+        if (this._thresholdHeightEnabled() && scroll >= this._getThresholdHeight()) {
+          event.gesture.stopDetect();
+
+          setImmediate(function() {
+            this._setState(this.STATE_ACTION);
+            this._translateTo(this._getHeight(), { animate: true });
+
+            this._waitForAction(this._onDone.bind(this));
+          }.bind(this));
+        }
+        else if (scroll >= this._getHeight()) {
+          this._setState(this.STATE_PREACTION);
+        }
+        else {
+          this._setState(this.STATE_INITIAL);
+        }
+  
+        this._translateTo(scroll);
+      },
+
+      _onDragStart: function(event) {
+        if (this.isDisabled()) {
+          return;
+        }
+
+        this._startScroll = this._getCurrentScroll();
+      },
+
+      _onDragEnd: function(event) {
+        if (this.isDisabled()) {
+          return;
+        }
+
+        if (this._currentTranslation > 0) {
+          var scroll = this._currentTranslation;
+
+          if (scroll > this._getHeight()) {
+            this._setState(this.STATE_ACTION);
+
+            this._translateTo(this._getHeight(), { animate: true });
+
+            this._waitForAction(this._onDone.bind(this));
+          }
+          else {
+            this._translateTo(0, { animate: true });
+          }
+        }
+      },
+
+      _waitForAction: function(done) {
+        if (this._attrs.ngAction) {
+          this._scope.$eval(this._attrs.ngAction, {$done: done});
+        }
+        else if (this._attrs.onAction) {
+          eval(this._attrs.onAction);
+        }
+        else {
+          done();
+        }
+      },
+
+      _onDone: function(done) {
+        this._safeApply();
+
+        this._translateTo(0, { animate: true });
+        this._setState(this.STATE_INITIAL);
+      },
+
+      _safeApply: function() {
+        var phase = this._scope.$root.$$phase;
+
+        if (phase !== '$apply' && phase !== '$digest') {
+          this._scope.$apply();
```

Comment:
Request: Don't refer $$phase to implement safeApply(). That is not recommended on AngularJS wiki(https://github.com/angular/angular.js/wiki/When-to-use-%24scope.%24apply%28%29). Use $timeout() or $scope.$evalAsync().